library/src/main/kotlin/com/connectrpc/protocols/ConnectInterceptor.kt

```diff
@@ -159,8 +160,7 @@ internal class ConnectInterceptor(
             streamResultFunction = { res ->
                 val streamResult: StreamResult<Buffer> = res.fold(
                     onHeaders = { result ->
-                        val responseHeaders =
-                            result.headers.filter { entry -> !entry.key.startsWith("trailer-") }
```

Comment:
This filtering was not correct. Only _unary_ RPCs use a "trailer-" suffix to distinguish headers and trailers. In streaming RPCs, everything in the response headers should be used as-is.